File: javascript/6_kyu/count_characters_in_your_string.js
// Count characters in your string
// https://www.codewars.com/kata/52efefcbcdf57161d4000091/train/javascript

function count(s) {
    if (s === "") {
        return {};
    } else {
        var ans = {};
        s = s.split("");
        for (var i = 0; i < s.length; i++) {
            if ( !(s[i] in ans) ) {
                ans[s[i]] = 1;
            } else {
                ans[s[i]]++;
            }
        }

        return ans;
    }
}

console.log(count('aba'));
console.log(count(''));

File: python/6_kyu/count_characters_in_your_string.py
# Count characters in your string
# https://www.codewars.com/kata/52efefcbcdf57161d4000091/train/python

def count(s):
    if s == "":
        return {}
    else:
        ans = {}
        s = list(s)
        for i in s:
            if i not in ans:
                ans[i] = 1
            else:
                ans[i] += 1
    
    return ans

# # Alternative, ideal answer
# from collections import Counter
# def count(s):
#     return Counter(s)

print(count('aba'))
print(count(''))